engine/src/vault/view.rs

```diff
@@ -183,6 +196,82 @@ impl<P: BoxProvider> DbView<P> {
     pub fn clear(&mut self) {
         self.vaults.clear();
     }
+
+    /// List the ids of all vaults.
+    pub fn list_vaults(&self) -> Vec<VaultId> {
+        self.vaults.keys().cloned().collect()
+    }
+
+    /// List the ids of all records in the vault.
+    pub fn list_records(&self, vid: &VaultId) -> Vec<RecordId> {
+        self.vaults
+            .get(vid)
+            .map(|v| v.entries.keys().map(|&c| c.into()).collect())
+            .unwrap_or_default()
+    }
+
+    /// List [`RecordId`] and [`BlobId`] of all entries in the vault.
```

Comment:
I wonder if its ok to expose the `BlobId`.  Before this was just an internal reference to connect the transaction head to the record; is this even useful for external resources?

Reply:
The `BlobId` is exposed so that we can check if the content of a record has changed, without needing to expose the actual secret. Each time we update its content, we create a new random `BlobId` for this record. This is useful for when we sync two vaults multiple times. To reduce overhead and risk, we only want to import the records that we don't have yet, or whose content has changed. For this we compare the `BlobId` on source and target, and only import the record if it is different.

Reply:
I was confused about the BlockId, too.
I think we should definitely think about renaming certain fields for clarity without breaking breaking the snapshot format